// node_templater/index.js
var fs = require('fs');
var p = require('path');
var TPL = {
	defaultFile: "default.html",
	temlateDir: "/templates/",
	/*
	future functons
	headerFile: "header",
	footerFile: 'footer'
	*/
	findFile: function (fileName) {
		//формируем путь
		var path = process.cwd()+ this.temlateDir + fileName + ".tpl.html";
		path = p.normalize(path);
		console.log(path);
		//проверяем есть ли
		var isFile = fs.existsSync(path); //true or false
		//__dirname текущая 
		//process.cwd() - корневой где запущен
		if (!isFile) {
			path = process.cwd()+ this.temlateDir + this.defaultFile;
			path = p.normalize(path);
		};
		//считываем файл
		/*read head adn read footer*/
		//file = head + body(fs.read)+footer; 
        return fs.readFileSync(path).toString();

	},
	replace: function (content, data) {
		var pattern, result = content;

		for (var i in data) {
			pattern = "%"+i+"%";
			pattern = new RegExp(pattern,'g');
			result = result.replace(pattern, data[i]);
		};
				return result;
	},
	init: function (tplName, data) {
		//tplName - имя файлы
		//data - объект данных
		//ищем файл tplName
		var file = this.findFile(tplName);
		// добавит данные в шаблон
		var content = this.replace(file, data);
		//console.log(file);
		return content;

	}
}

module.exports = TPL;
