// route_handler/index.js
var DB = require("node_server_db");
var TPL = require("node_templater");
DB.init("app");
var route_handler = {
	index: function(callback, router){
		//DB.insert("users",{name: 'user1', login: 'user'}, function(a,b){
		//	console.log('ok', b);
		//});
		//получит данные из БД
		//эти данные передать в шаблон (название и данные)
		/*
		{
			title:"",
			body: "",
			atr: ""
		}
		*/
		//полученные данные выводим в поток 
		//закрываем соединение
		var data = {
			title: "Главная страница",
			body: "<h1>На главной</h1>"

		}
		var content = TPL.init('c',data);		
		router.write(content);
		router.end();
	},
	contacts: function(callback, router){
		callback("contacts");
		router.end();
	},
	rest: function(callback, router){
		DB.select("users", {}, function(err, res){
			router.write(JSON.stringify(res));
			router.end();
		});
		
	},
	delete: function(callback, router){

		var filter = {name: "user2"};

		DB.delete("users", filter, (err, res) => {
			if (err) {
				router.write('sorry some error');
			} else {
				router.write("Deleted: " + res.deletedCount + " elements");
			}
			router.end();
		});
		
	}
	,
	update: function(callback, router){

		var filter = {name: "user3"};
		var data =  {name: "user3=>user4"};
		
		DB.update("users", filter, data, (err, res) => {
			if (err) {
				router.write('sorry some error');
			} else {
				console.log(res);
				router.write("updated");

			}
			router.end();
		});
		
	}
};

module.exports = route_handler;
